persist/fs/persist_manager.go

```diff
@@ -115,20 +182,40 @@ func (pm *persistManager) reset() {
 }
 
 func (pm *persistManager) Prepare(namespace ts.ID, shard uint32, blockStart time.Time) (persist.PreparedPersist, error) {
+	pm.Lock()
+	currMetrics, exists := pm.metrics[shard]
+	if !exists {
+		currMetrics = newShardMetrics(pm.scope, shard)
+		pm.metrics[shard] = currMetrics
+	}
+	currMetrics.reset()
+	pm.currMetrics = currMetrics
+	pm.Unlock()
+
+	// NB(xichen): explicitly not using a defer to avoid creating lambdas on
+	// the heap and causing additional GC
+	start := pm.nowFn()
 	var prepared persist.PreparedPersist
 
 	// NB(xichen): if the checkpoint file for blockStart already exists, bail.
 	// This allows us to retry failed flushing attempts because they wouldn't
 	// have created the checkpoint file.
 	if FilesetExistsAt(pm.filePathPrefix, namespace, shard, blockStart) {
+		end := pm.nowFn()
+		currMetrics.recordWorked(end.Sub(start))
 		return prepared, nil
 	}
 	if err := pm.writer.Open(namespace, shard, blockStart); err != nil {
+		end := pm.nowFn()
+		currMetrics.recordWorked(end.Sub(start))
 		return prepared, err
 	}
 
 	prepared.Persist = pm.persist
 	prepared.Close = pm.close
+
+	end := pm.nowFn()
+	currMetrics.recordWorked(end.Sub(start))
```

Comment:
Is it worth reporting any stats for checking exists and opening?  Perhaps we just want to look at the actual file writing/throughput?